origin-discovery/src/lib/search.js

```diff
@@ -51,11 +52,20 @@ class Listing {
    * @returns The listingId indexed.
    */
   static async index (listingId, buyerAddress, ipfsHash, listing) {
+    // make a copy of the ipfs data
```

Comment:
Why do we need to make a copy via the ipfs data rather than copying listing directly ? e.g.:
`
const originListing = JSON.parse(JSON.stringify(listing))
originListing.commission = originListing.commissionUsedToBoostListing
`